src/discovery/graphql-root-tx-index.ts

```diff
@@ -243,4 +284,73 @@ export class GraphQLRootTxIndex implements DataItemRootTxIndex {
 
     return NOT_FOUND;
   }
+
+  private async queryItemMetadata(
+    id: string,
+    log: winston.Logger,
+  ): Promise<
+    | {
+        contentType?: string;
+        size?: string;
+      }
+    | typeof NOT_FOUND
+  > {
+    // lower number = higher priority
+    const priorities = Array.from(this.trustedGateways.keys()).sort(
+      (a, b) => a - b,
+    );
+
+    let lastError: Error | null = null;
+
+    for (const priority of priorities) {
+      const gatewaysInTier = this.trustedGateways.get(priority);
+
+      if (gatewaysInTier) {
+        const shuffledGateways = shuffleArray([...gatewaysInTier]);
+
+        for (const gatewayUrl of shuffledGateways) {
+          try {
+            const response = await this.axiosInstance.post(
+              `${gatewayUrl}/graphql`,
+              {
+                query: GRAPHQL_QUERY,
```

Comment:
I think we should split this out into a separate query so we don't need to retrieve the metadata when doing the root parent traversal.